### PR TITLE
SWATCH-253: Add support for Kafka over TLS and using SASL

### PR DESCRIPTION
**Pull request by awood**
# Testing
* You'll need to destroy your existing `kafka` pod and recreate it with the new TLS configuration.  Myself, I have just been bringing everything down with `podman-compose down`, then `podman-compose up -d`.  This series of commands will wipe your dev database though.  Either do a `pg_dump` before `podman-compose down` and do an import after `podman-compose up`.  *OR* `podman stop` and `podman rm` the `rhsm-subscriptions_kafka_1` and run `podman-compose up` again.  I believe that will be sufficient to just recreate the Kafka container.  I would appreciate verification of this though.  It seems to work for me after some quick testing.
* Start the application and tell it to connect to Kafka over TLS using SASL (with some extra logging) 
  ```
  KUBERNETES_SERVICE_HOST="f" \
  KUBERNETES_SERVICE_PORT="f" \
  ACG_CONFIG=./config/kafka/clowder-config-test.json \
   ./gradlew :bootRun --args="--logging.level.org.apache.kafka.common.security.ssl=DEBUG \
  --logging.level.org.apache.kafka.common.network.SslTransportLayer=DEBUG \
  --logging.level.org.apache.kafka.common.security.authenticator.SaslClientAuthenticator=DEBUG"
  ```
* You should see the TLS connection get established and the SASL authentication succeed.
* Ensure TLS settings are not loaded for dev:
  `./gradlew :bootRun`
* Reserve a Clowder namespace and mark it as using managed Kafka:
  ```
  bonfire namespace reserve -d 240h --pool managed-kafka
  ```
* Apply this patch
```patch 
diff --git a/swatch-core/src/main/resources/swatch-core/application.yaml b/swatch-core/src/main/resources/swatch-core/application.yaml
index 80f6f55a..462ba81b 100644
--- a/swatch-core/src/main/resources/swatch-core/application.yaml
+++ b/swatch-core/src/main/resources/swatch-core/application.yaml
@@ -60,6 +60,24 @@ DATABASE_MAX_POOL_SIZE: 10
 server:
   port: ${SERVER_PORT}
 
+logging:
+  level:
+    org:
+      apache:
+        kafka:
+          common:
+            network:
+              SslTransportLayer: DEBUG
+            security:
+              authenticator:
+                SaslClientAuthenticator: DEBUG
+          clients:
+            consumer:
+              ConsumerConfig: DEBUG
+      springframework:
+        boot:
+          context: DEBUG
+
 management:
   server:
     port: ${METRICS_SERVER_PORT}
```
* Commit the above with a temp commit (e.g. "WIP") and perform a build (the script below will push the images to your personal quay.io repo, so login there first.
  ```
  $ bin/build-images.sh
  ```
* Do a deploy to the ephemeral environments, replacing `YOUR_QUAY_USER` with your quay.io username:
  ```
  $ oc project YOUR_ALLOCATED_NAMESPACE_HERE # retrieve with bonfire namespace list --mine
  $ bonfire deploy --pool managed-kafka -n $(oc project -q) --no-remove-resources=swatch-producer-aws --no-remove-resources=rhsm-subscriptions --optional-deps-method=none -p rhsm-subscriptions/IMAGE=quay.io/YOUR_QUAY_USER/rhsm -i quay.io/YOUR_QUAY_USER/rhsm=$(git rev-parse --short=7 HEAD) -p rhsm-subscriptions/CONDUIT_IMAGE=quay.io/YOUR_QUAY_USER/swatch-system-conduit -i quay.io/YOUR_QUAY_USER/swatch-system-conduit=$(git rev-parse --short=7 HEAD) -p swatch-producer-aws/IMAGE=quay.io/YOUR_QUAY_USER/swatch-producer-aws -i quay.io/YOUR_QUAY_USER/swatch-producer-aws=$(git rev-parse --short=7 HEAD) rhsm
  ```

# TODO
- [ ] Send a note to all devs when this merges indicating that they will need to recreate their `kafka` pod.

### CODE REVIEW

**Comment by kahowell** [comment]
FYI, failing in EE w/

```
***************************
APPLICATION FAILED TO START
***************************

Description:

Failed to bind properties under 'spring.kafka.jaas.options.username' to java.lang.String:

    Property: spring.kafka.jaas.options.username
    Value: "${KAFKA_USERNAME}"
    Origin: class path resource [swatch-core/application.yaml] from rhsm-subscriptions-1.1.0-snapshot.202208052011.uncommitted+a57d8ea.jar!/BOOT-INF/lib/swatch-core-1.1.0-snapshot.202208052011.uncommitted+a57d8ea.jar - 208:19
    Reason: java.lang.IllegalArgumentException: Could not resolve placeholder 'clowder.kafka.brokers[0].sasl.username' in value "${clowder.kafka.brokers[0].sasl.username}"

Action:

Update your application's configuration
```

**Comment by awood** [comment]
Which pod was failing with that error, @kahowell?

**Comment by ntkathole** [comment]
> Which pod was failing with that error, @kahowell?

@awood I see Job build artifacts has rhsm-subscriptions-worker-648ff977f4-kdzwl_rhsm-subscriptions-worker-init.log file showing above error.

**Comment by awood** [comment]
I made a push that at least fixes the issue with the init pods, but the ultimate issue looks like there's no `cdappconfig.json` file being read during the Jenkins run.  @kahowell Let's talk Monday and see if we can figure this out.

File: deploy/clowder.md [hunk 1]
```diff
@@ -13,206 +13,90 @@
...
-service using `port-forward`.  For example: `kubectl port-forward svc/rhsm-api 8000`
-will forward your localhost's port 8000 to the service (and underlying pod).
-
-## Ephemeral Development
-This approach is the recommended one.  The definitive reference is going to be
-the "Onboarding to the Ephemeral Cluster" page in the Clouddout documentation,
-but here are some essentials:
+### Ephemeral Development and Deployment
+The definitive reference is going to be the "Onboarding to the Ephemeral Cluster" page in the 
+Clouddout documentation, but here are some essentials:
```
Comment (kahowell) [inline deploy/clowder.md:99]: Typo: Clouddout; at first I read this as "cloud-doubt" :joy: .

File: deploy/clowder.md [hunk 2]
```diff
@@ -13,206 +13,90 @@
...
+
+    - name: swatch-producer-aws
+      host: local
+      repo: $(pwd)/rhsm-subscriptions/swatch-producer-aws
+      path: /deploy/clowdapp.yaml
+      parameters:
+        REPLICAS: 1
+        swatch-producer-aws/IMAGE: quay.io/cloudservices/swatch-producer-aws
+BONFIRE
+```
```
Comment (kahowell) [inline deploy/clowder.md:95]: Mind to correct the path in swatch-api, while you are making changes?

**Review by kahowell — CHANGES_REQUESTED** [review]
Overall looks good, left a couple inline comments; also can you please s/dummy/placeholder in
* `config/kafka/client.properties`
* `config/kafka/clowder-config-test.json`
* `config/kafka/kafka_server_jaas.conf`
* `swatch-core/src/main/resources/swatch-core/application.yaml`
?

**Review by kahowell — APPROVED** [review]

**Comment by kahowell** [comment]
failing test: `tests.e2e.test_ephemeral_conduit.test_e2e_conduit_ephermal` unrelated

**Pull request merged**